File: ios/Theme/ResultAnimation.swift
public class ResultAnimation: ViewStyle {
  private static let KEY = "resultAnimation"
  private let style: Style
  private let target: NSDictionary?
  private let image: Image
  private let color: Color
  
  init(target: NSDictionary?) {
    self.style = Style()
    self.color = Color()
    
    self.target = self.style.getTarget(target, key: ResultAnimation.KEY)
    
    self.image = Image(target: self.target)
    
    super.init(target: target, key: ResultAnimation.KEY)
  }

  override public func getBackgroundColor() -> UIColor {
    return self.getBackgroundColor(defaultColor: "#026ff4")
  }
  
  public func getDisplayTime() -> Double {
    let defaultDisplayTime: Double = 2.5
    let key = "displayTime"
    
    if !self.style.exists(self.target, key: key) {
      return defaultDisplayTime
    }
    
    let displayTime = self.target?[key] as! Double
    return displayTime < 0 ? defaultDisplayTime : displayTime
  }
  
  public func getIDScanSuccessForegroundColor() -> UIColor {
    return self.color.getColor(self.target, key: "idScanSuccessForegroundColor", defaultColor: "026ff4")
  }
  
  public func getUnsuccessBackgroundColor() -> UIColor {
    return self.color.getColor(self.target, key: "unsuccessBackgroundColor", defaultColor: "CC0044")
  }
  
  public func getUnsuccessForegroundColor() -> UIColor {
    return self.color.getColor(self.target, key: "unsuccessForegroundColor")
  }
  
  public func getSuccessBackgroundImage() -> UIImage? {
    return self.image.getSource("successImage")
  }
  
  public func getUnsuccessBackgroundImage() -> UIImage? {
    return self.image.getSource("unsuccessImage")
  }
}

File: android/src/main/java/com/azifacemobile/theme/ResultAnimation.java
package com.azifacemobile.theme;

import com.azifacemobile.utils.Theme;
import com.azifacemobile.theme.abstracts.ViewStyle;
import com.facebook.react.bridge.ReactApplicationContext;

import org.json.JSONObject;

public class ResultAnimation extends ViewStyle {
  private static final String KEY = "resultAnimation";
  private final JSONObject target;
  private final Image image;
  private final Color color;

  public ResultAnimation(ReactApplicationContext context, JSONObject target) {
    super(target, KEY);

    this.target = new Theme().getTarget(target, KEY);

    this.color = new Color();
    this.image = new Image(context, this.target);
  }

  @Override
  public int getBackgroundColor() {
    return super.getBackgroundColor("#026ff4");
  }

  public int getIDScanSuccessForegroundColor() {
    return this.color.getColor(this.target, "idScanSuccessForegroundColor", "#026ff4");
  }

  public int getUnsuccessBackgroundColor() {
    return this.color.getColor(this.target, "unsuccessBackgroundColor", "#CC0044");
  }

  public int getUnsuccessForegroundColor() {
    return this.color.getColor(this.target, "unsuccessForegroundColor");
  }

  public int getSuccessBackgroundImage() {
    return this.image.getSource("successImage");
  }

  public int getUnsuccessBackgroundImage() {
    return this.image.getSource("unsuccessImage");
  }
}
